src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -408,7 +408,7 @@ protected virtual void OutputVirtualSlots(NodeFactory factory, ref ObjectDataBui
             if (declType.HasGenericDictionarySlot())
             {
                 // Note: Canonical type instantiations always have a generic dictionary vtable slot, but it's empty
-                if (declType.IsCanonicalSubtype(CanonicalFormKind.Any))
+                if (declType.IsInterface || declType.IsCanonicalSubtype(CanonicalFormKind.Any))
```

Comment:
I'll put a TODO comment to generate the right dictionary entry for static methods.